cmd/cli/serve/util.go

```diff
@@ -168,9 +168,10 @@ func SetupIPFSClient(ctx context.Context, cm *system.CleanupManager, ipfsCfg typ
 	return client, nil
 }
 
+
 func StartWebUIServer(ctx context.Context) error {
 	// Define command to start React server
-	cmd := exec.Command("npm", "run", "build")
+	cmd := exec.CommandContext(ctx, "npm", "run", "build")
```

Comment:
Isn't this something that should happen in the Makefile as a build step, and then the output of that could be embedded in the binary and served from here over http?